testing/demo/pkg/transfer/relay.go

```diff
@@ -0,0 +1,261 @@
+package main
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+
+	proverclient "github.com/celestiaorg/celestia-zkevm-ibc-demo/provers/client"
+	"github.com/celestiaorg/celestia-zkevm-ibc-demo/testing/demo/pkg/ethereum"
+	"github.com/celestiaorg/celestia-zkevm-ibc-demo/testing/demo/pkg/utils"
+	coretypes "github.com/cometbft/cometbft/rpc/core/types"
+	transfertypes "github.com/cosmos/ibc-go/v10/modules/apps/transfer/types"
+	"github.com/cosmos/solidity-ibc-eureka/abigen/ics26router"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// FungibleTokenPacketData represents the ICS20 token transfer data structure
+// It should match IICS20TransferMsgs.FungibleTokenPacketData in the Solidity contract
+type FungibleTokenPacketData struct {
```

Comment:
This isn't already somewhere we can re-use?

Reply:
There is a Go struct for it https://github.com/rootulp/ibc-go/blob/ac0297b3f35e33076732afeedbd3a5dd8f07cc53/modules/apps/transfer/types/packet.pb.go#L26-L40

But there isn't a Go struct for the representation the Solidity expects.